server/controllers/authController.js

```diff
@@ -168,36 +108,94 @@ const handleLogin = async (req, res) => {
   }
 };
 
-//@endpoint GET /user/profile
-//@desc authenticate user token and send back user info
-//@access PRIVATE
+// ----------------------------------------SIGNUP----------------------------------------
+//@endpoint POST /SignUp
+//@desc Sign up user info.
+//@access PUBLIC
 
-const handleGetUser = async (req, res, next) => {
-  //from middleware.
-  let id = req.user.id;
-  try {
-    const db = getConnection().db(dbName);
-    //see if you find the user.
-    let checkForUser = await db
-      .collection(collectionUsers)
-      .findOne({ _id: ObjectId(id) });
-    res.status(200).json({
-      status: 200,
-      message: "User remains signed in. Token verification accepted.",
-      username: checkForUser.username,
-      accessToken: checkForUser.accessToken,
-      _id: checkForUser._id,
-      profileImage: checkForUser.profileImage,
+const handleSignUp = async (req, res) => {
+  //get from body.
+  let filePath = req.file.path;
+  let name = req.body.name;
+  let pass = req.body.pass;
+  //initialize into new var.
+  let signUpInfo = {
+    user: name,
+    pass: pass,
+  };
+  //check if any of the fields are empty.
+  if (signUpInfo.user && signUpInfo.pass && req.file) {
+    //logic for pass hashing.
+    const saltRounds = 10;
+    let register = new Date();
+    let hashedPass;
+    bcrypt.hash(signUpInfo.pass, saltRounds, function (err, hash) {
+      if (err) throw err;
+      hashedPass = hash;
+      // Store hash in your password DB.
+    });
+    try {
+      //see if you find the user. //check for existing user
+      let checkForUser = await getUserByUserName(signUpInfo.user);
+      if (!checkForUser) {
+        let information = {
+          username: signUpInfo.user,
+          password: hashedPass,
+          registrationDate: register,
+          profileImage: filePath,
+        };
+        let r = await insertNewUser(information);
+        let userId = r.ops[0]._id;
+        let getUser = await getUserByUserName(signUpInfo.user);
+        //give the same _id of the user to the collectionUserEvents.
+        //to keep track of all the events a user is participating in.
+        //This way it is relational.
+        let userEvent = await createUserEvent(userId);
```

Comment:
This should be a separate function, for readability. If user doesn't exist, you should call a `createUser` method in the service. Also, an insert operation returns the whole created value on `result.ops`, so no need to do the second `getUserByUserName`

Reply:
my createuser method is the insertNewUser method.
and absolutely right about the second getUserByUserName!
thanks